numpy/core/arrayprint.py

```diff
@@ -521,6 +509,8 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
+    if style is not None and a.shape == () and a.dtype.names is None:
+        return style(a[()])
```

Comment:
OK, so this doesn't affect any `style=None` option in an explicit `array2string`, correct? Do agree that `str(array_scalar_string)` should give a fixed result, which is just the string/unicode/bytes object.